libs/partners/ai21/langchain_ai21/semantic_text_splitter.py

```diff
@@ -0,0 +1,132 @@
+import copy
+import logging
+import re
+from typing import (
+    Any,
+    Iterable,
+    List,
+    Optional,
+)
+
+from ai21.models import DocumentType
+from langchain_core.documents import Document
+from langchain_text_splitters import TextSplitter
+
+from langchain_ai21.ai21_base import AI21Base
+
+logger = logging.getLogger(__name__)
+
+
+class AI21SemanticTextSplitter(TextSplitter):
+    """Splitting text into coherent and readable units,
+    based on distinct topics and lines
+    """
+
+    class Config:
+        """Configuration for this pydantic object."""
+
+        arbitrary_types_allowed = True
+
+    def __init__(
+        self,
+        chunk_size: int = 0,
+        chunk_overlap: int = 0,
+        client: Optional[Any] = None,
+        **kwargs: Any,
```

Comment:
Can you please add the following args to the `__init__` ? since we can't inherit from `AI21Base` we're missing other params 
```python
api_key: Optional[SecretStr] = None
api_host: Optional[str] = None
timeout_sec: Optional[float] = None
 num_retries: Optional[int] = None
```